File: src/main/java/ru/nucodelabs/gem/view/MainViewModule.java
package ru.nucodelabs.gem.view;

import com.google.common.eventbus.EventBus;
import com.google.inject.AbstractModule;
import ru.nucodelabs.gem.model.Section;
import ru.nucodelabs.gem.model.SectionImpl;
import ru.nucodelabs.gem.view.charts.MisfitStacksController;
import ru.nucodelabs.gem.view.charts.VESCurvesController;
import ru.nucodelabs.gem.view.main.MainViewController;
import ru.nucodelabs.gem.view.main.PicketsBarController;
import ru.nucodelabs.gem.view.tables.ExperimentalTableController;
import ru.nucodelabs.gem.view.tables.ModelTableController;

import static com.google.inject.Scopes.SINGLETON;

public class MainViewModule extends AbstractModule {
    @Override
    protected void configure() {
        bind(Section.class).to(SectionImpl.class).in(SINGLETON);
        bind(EventBus.class).in(SINGLETON);
        bind(MainViewController.class);
        bind(VESCurvesController.class);
        bind(MisfitStacksController.class);
        bind(ExperimentalTableController.class);
        bind(ModelTableController.class);
        bind(PicketsBarController.class);
    }
}

File: src/main/java/ru/nucodelabs/gem/view/main/MainViewModule.kt
package ru.nucodelabs.gem.view.main

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.google.inject.Singleton
import ru.nucodelabs.gem.fxmodel.ObservableSection
import ru.nucodelabs.geo.ves.Section
import ru.nucodelabs.gem.app.io.StorageManager
import ru.nucodelabs.gem.app.snapshot.HistoryManager
import ru.nucodelabs.gem.app.snapshot.Snapshot
import ru.nucodelabs.gem.view.DialogsModule
import ru.nucodelabs.gem.view.charts.ChartsModule

class MainViewModule : AbstractModule() {
    override fun configure() {
        bind(MainViewController::class.java).`in`(Singleton::class.java)
        bind(StorageManager::class.java).`in`(Singleton::class.java)
        bind(ObservableSection::class.java).`in`(Singleton::class.java)

        install(DialogsModule())
        install(ObservableDataModule())
        install(ChartsModule())
    }

    @Provides
    @Singleton
    fun sectionHistoryManager(sectionOriginator: Snapshot.Originator<Section>): HistoryManager<Section> =
        HistoryManager(sectionOriginator)

    @Provides
    fun sectionOriginator(observableSection: ObservableSection): Snapshot.Originator<Section> = observableSection

    @Provides
    fun fileImporter(mainViewController: MainViewController): FileImporter = mainViewController

    @Provides
    fun fileOpener(mainViewController: MainViewController): FileOpener = mainViewController
}
